pkg/flow/adapter/synchronizer/ce.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synchronizer
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2/event"
+)
+
+// Correlation Key parameters.
+const (
+	defaultCorrelationKeyLen = 24
+	maxCorrelationKeyLen     = 64
+
+	correlationKeycharset = "abcdefghijklmnopqrstuvwxyz0123456789"
+)
+
+var (
+	// CloudEvent attributes cannot be used as a correltaion key.
+	restrictedKeys = []string{
+		"id",
+		"type",
+		"time",
+		"subject",
+		"schemaurl",
+		"dataschema",
+		"specversion",
+		"datamediatype",
+		"datacontenttype",
+		"datacontentencoding",
+	}
+
+	seededRand *rand.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+)
+
+// CloudEventKey is the Correlation Key for the CloudEvents context.
+type CloudEventKey struct {
+	Attribute string
+	Length    int
+}
+
+// NewCorrelationKey returns an instance of the CloudEvent Correlation key.
+func NewCorrelationKey(attribute string, length int) (*CloudEventKey, error) {
```

Comment:
nit: the file is called `ce` the type `CloudEventKey` and the constructor `NewCorrelationKey`. I find it a bit arbitrary.
the function is not returning a CE key nor a correlation key, it looks like some sort of manager ...

Reply:
https://github.com/triggermesh/triggermesh/pull/353/commits/c4b31e57f8f79488eb22f3b05b654f75d36e9bd8